### PR TITLE
Remove deprecated CupertinoDialog

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This removes the deprecated `CupertinoDialog` widget that has reached the EOL . The `CupertinoPopupSurface` or `CupertinoAlertDialog` should be used instead.

Part of deprecations that are slated for removal for next release in https://flutter.dev/go/deprecation-lifetime

More context: https://medium.com/flutter/deprecation-lifetime-in-flutter-e4d76ee738ad

## Related Issues

Deprecated in https://github.com/flutter/flutter/pull/20649
https://github.com/flutter/flutter/issues/73910

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
@goderbauer I know we can't test the fix, but we're going to be removing the deprecation anyways, what do you think about including the fix support?

File: packages/flutter/lib/fix_data.yaml [hunk 1]
```diff
@@ -11,6 +11,33 @@
 version: 1
 transforms:
 
+  # Change made in https://github.com/flutter/flutter/pull/20649
+  # TODO(Piinks): Add tests when `bulkApply:false` testing is supported.
```
Comment (goderbauer) [inline packages/flutter/lib/fix_data.yaml:15]: Is there an issue on "dart fix" to add testing support for this that we can link from here?

Comment (goderbauer) [reply packages/flutter/lib/fix_data.yaml:15]: And do we know what the timeline for adding that support is?

Comment (Piinks) [reply packages/flutter/lib/fix_data.yaml:15]: Filed https://github.com/dart-lang/sdk/issues/44639

Comment (Piinks) [inline packages/flutter/lib/fix_data.yaml:15]:
```suggestion
  # TODO(Piinks): Add tests when `bulkApply:false` testing is supported, https://github.com/dart-lang/sdk/issues/44639
```

File: packages/flutter/lib/fix_data.yaml [hunk 2]
```diff
@@ -11,6 +11,33 @@
...
+    bulkApply: false
+    element:
+      uris: [ 'cupertino.dart' ]
+      class: 'CupertinoDialog'
+    changes:
+      - kind: 'rename'
+        newName: 'CupertinoPopupSurface'
+
+  # Change made in https://github.com/flutter/flutter/pull/20649
+  # TODO(Piinks): Add tests when `bulkApply:false` testing is supported.
```
Comment (Piinks) [inline packages/flutter/lib/fix_data.yaml:27]:
```suggestion
  # TODO(Piinks): Add tests when `bulkApply:false` testing is supported, https://github.com/dart-lang/sdk/issues/44639
```

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
Confirmed the google test failures here were unrelated and everything passes. Manually set to green.

**Pull request merged**